colossalai/shardformer/layer/utils.py

```diff
@@ -53,30 +58,35 @@ def allreduce_partial_data_grad(tp_group: ProcessGroup, model: nn.Module = None,
         assert (model is not None) ^ (grads is not None), "Exactly one of model and grads must be not None."
 
         # Get the size of the process group, which determines whether synchronization is needed.
-        tp_size = get_world_size(tp_group) if tp_group is not None else 1
+        group_size = get_world_size(process_group) if process_group is not None else 1
 
-        if tp_size == 1:
+        if group_size == 1:
             # If the process group size is 1, no synchronization is required.
             return
 
         if model is not None:
             # If `model` is provided, extract partial derived gradients from the model's parameters.
             grads = []
             for p in model.parameters():
-                if p.grad is not None and SeqParallelUtils.is_sp_partial_derived_param(p):
-                    grads.append(p.grad.data)
+                if p.grad is not None:
+                    if require_flag and SeqParallelUtils.is_sp_partial_derived_param(p):
+                        grads.append(p.grad.data)
+                    elif not require_flag:
```

Comment:
Why append grad even if require_flag is false?

Reply:
fixed